### DISCUSSION TITLE
S0 Zähler einrichten

### DISCUSSION THREAD

**Discussion by bebu259** [camueller/SmartApplianceEnabler#466] [Installation und Betrieb]
Hallo 👋,  ich habe sein kurzem eine SMA Anlage mit Sunny Home Manager und bin auf der Recherche, wie ich Zähler einbinden kann, auf dieses Projekt gestoßen, wirklich großartig @camueller !

Ich habe einen Pi4B mit der _Standard-Installation_ eingerichtet (mit dem Latest Image 2022.01.28-lite).

Nun versuche einen S0 Zähler (B+G DRS155DC) einzubinden: der Zähler lässt sich im Sunny Portal als Gerät anmelden, zeigt aber immer 0 Watt an (wobei das Display des Zählers 60 Watt anzeigt).

Ich habe in [dieser Diskussion](https://github.com/camueller/SmartApplianceEnabler/discussions/230) gesehen, dass schon andere dieses Problem hatten.

Wenn ich die Ports mit `gpio readall` testen will, bekomme ich einen `-bash: gpio: command not found` Fehler.

@camueller, du hattest [hier](https://github.com/camueller/SmartApplianceEnabler/discussions/230#discussioncomment-3363188) erwartet, dass der Befehl fehlschlägt...

Muss ich noch etwas manuell installieren?

**Comment by camueller** [comment]
Der `gpio`-Befehl gehört zu `wiringpi`, das auf den aktuellen Raspberry Pi OS-Images nicht mehr vorhanden ist und auch vom SAE aber Version 2.x nicht mehr verwendet wird.

Zur gestellten Frage finden sich hier Hinweise zur Fehlersuche: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md#semp2

**Reply by bebu259** [reply]
Danke für die schnelle Antwort!

Die `AveragePower` ist tatsächlich 0:

```xml
<DeviceStatus>
  <DeviceId>F-20230209-123456000001-00</DeviceId>
  <EMSignalsAccepted>false</EMSignalsAccepted>
  <Status>On</Status>
  <PowerConsumption>
    <PowerInfo>
      <AveragePower>0</AveragePower>
      <Timestamp>0</Timestamp>
      <AveragingInterval>60</AveragingInterval>
    </PowerInfo>
  </PowerConsumption>
</DeviceStatus>
```

Im Log steht für den Zähler auch 0 (obwohl der Zähler selber auf seinem Display 60 Watt anzeigt):

```
2023-02-09 20:44:44,650 TRACE [pool-11-thread-1] d.a.s.m.MqttClient [MqttClient.java:265] F-20230209-123456000001-00-MQTT-S0ElectricityMeter: Publish message: topic=sae/F-20230209-123456000001-00/Meter payload={"energy":0.0,"power":0,"time":"2023-02-09T20:44:44.649616","type":"MeterMessage"}
2023-02-09 20:44:51,408 TRACE [Timer-0] d.a.s.u.GuardedTimerTask [GuardedTimerTask.java:54] F-19690925-481470000001-00: Executing timer task name=MqttPublish-S0ElectricityMeter id=10321031
2023-02-09 20:44:51,409 TRACE [pool-11-thread-1] d.a.s.m.MqttClient [MqttClient.java:265] F-19690925-481470000001-00-MQTT-S0ElectricityMeter: Publish message: topic=sae/F-19690925-481470000001-00/Meter payload={"energy":0.0,"power":0,"time":"2023-02-09T20:44:51.408701","type":"MeterMessage"}
```

Wie könnte ich prüfen, ob die S0-Impulse an den PINS ankommen?

Ich weiß nicht, ob das relevant ist, aber ich habe bei _Min. Leistungsaufnahme_ diese Werte eingetragen:

<img width="599" alt="Bildschirmfoto 2023-02-09 um 20 31 24" src="https://user-images.githubusercontent.com/1553063/217918530-663c8580-a0a1-47d7-bace-475da50a8939.png">

**Reply by bebu259** [reply]
Der Zähler läuft jetzt!

Ich hatte nicht bedacht, dass bei 60 Watt und 1.000 Impulsen / 1 kWh nur alle 60 Sekunden ein Impuls kommt und solange hatte ich gestern nie abgewartet.

Eine sinnvolle Wattzahl bei 60 Watt kann ja überhaupt erst nach dem Eingang von 2 Impulsen, also allerfrühestens nach einer Minute angezeigt werden, und genau so ist es auch:

<img width="1373" alt="Bildschirmfoto 2023-02-10 um 13 11 07" src="https://user-images.githubusercontent.com/1553063/218099023-7d15f47a-08b1-43c7-b2e7-ec73fc27b3cb.png">

Danke für deine Hilfe und für dieses tolle System!

**Reply by camueller** [reply]
Genauso ist es - dem kann ich nichts hinzufügen.
